### PR TITLE
Rubrics assessed view VDiff +assessment data dump - US135302

### PR DESCRIPTION
**Pull request by imcdonald-d2l**
It's a big one... got lots of data attached for the rubrics assessment view.

[Rally ticket](https://rally1.rallydev.com/#/57732444928d/custom/373260458992?detail=%2Fuserstory%2F622541396279)

### CODE REVIEW

File: src/components/rubric.js [hunk 1]
```diff
@@ -357,27 +357,31 @@ class Rubric extends composeMixins(
...
+			this._assessmentEntityChanged();
 		}
 	}
 
 	isCompact() {
 		return this.forceCompact || this._isMobile;
 	}
 
+	_assessmentEntityChanged() {
+		this._assessedActivityName = this._assessmentEntity.name;
```
Comment (imcdonald-d2l) [inline src/components/rubric.js]: Moved this to a separate function so I could manually call this in vdiff files. No logic has been changed. Vdiff didn't seem to be going through the `update(changedProperties)` lifecycle when it should have.

File: src/components/rubric.js [hunk 2]
```diff
@@ -451,7 +455,7 @@ class Rubric extends composeMixins(
 
 	_hasOverallScore(entity, compact, compactOverallScore) {
 		const isCompact = !compact || compactOverallScore;
-		return entity.overallLevelsRel?.href && !entity.isHolistic && isCompact;
+		return entity?.overallLevelsRel?.href && !entity?.isHolistic && isCompact;
```
Comment (imcdonald-d2l) [inline src/components/rubric.js]: All the `?` were added because vdiff was complaining.

File: test/perceptual/utilities.js [hunk 3]
```diff
@@ -27,6 +27,23 @@ export const waitForEntityReady = async(page, selector) => {
 	}, selector);
 };
 
+export const callAssessmentEntityChanged = async(page, selector) => {
+	await page.evaluate(selector => {
+		return new Promise(resolve => {
+			const element = document.querySelector(selector);
+			element._assessmentEntityChanged();
```
Comment (imcdonald-d2l) [inline test/perceptual/utilities.js]: For some reason the `updated()` lifecycle method wasn't being called in vdiff tests only. Vdiff demo files were working just fine. Added this as a hack to get assessment entity properties updated for vdiff.

**Comment by imcdonald-d2l** [comment]
The `pop-out-assessed-renders-pop-out-points-assessed` seems to have been flaky in the past where it just wouldn't render the rubric. Doesn't happen locally though. I increased the timeout and it doesn't seem to happen. If it is flaky in the future I'm hoping increasing the timeout in the `callAssessmentEntityChanged` fixes things. 🤞🏻

**Review by devpow112 — APPROVED** [review]
Looks good to me :tada:

**Pull request merged**